### PR TITLE
Express attempt 2

### PR DESCRIPTION
**Pull request by Gisleburt**
I think we should discuss me tactics here. 

There will be a little more work to deal with apps with more complext state, but at worst we can dump it into the html and read it back in.

### CODE REVIEW

File: srv/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// @flow
+require('@babel/register')(require('../helpers/babel.config'));
```
Comment (Gisleburt) [inline srv/index.js]: Note the reuse of `babel.config.js`

File: srv/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+// We're ok to block here because we don't want the server to start until we have this file and
+// we only want to do this once ever.
+const template = fs
+  .readFileSync(path.join(webpackConfig.output.path, 'index.html'))
+  .toString();
+
+const app = express();
+
+const render = ({ url }) => {
```
Comment (Gisleburt) [inline srv/index.js]: We can use this function to pass various options through to the App. At the moment I'm just passing url

File: package.json [hunk 3]
```diff
@@ -37,7 +38,9 @@
   "homepage": "https://github.com/apolitical/frontend-skeleton#readme",
   "dependencies": {
     "@babel/polyfill": "^7.2.5",
+    "@babel/register": "^7.0.0",
     "core-decorators": "^0.20.0",
+    "express": "^4.16.4",
```
Comment (Gisleburt) [inline package.json]: Probably a bunch of things need to move out of dev dependencies

File: srv/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  const reactApp = appBase({
+    url,
+    ssr: true,
+  });
+  const renderdString = ReactDomServer.renderToString(reactApp);
+  return template.replace(marker, marker + renderdString);
+};
+
+app.use(express.static(buildDir));
+app.get('/*', (req, res) => res.send(render(req)));
```
Comment (Gisleburt) [inline srv/index.js]: Should probably add a path for favicon... or just add a favicon?

Comment (huwd) [reply srv/index.js]: do we have a favicon?

File: src/base.jsx [hunk 5]
```diff
@@ -14,13 +19,23 @@ const reducers = combineReducers({
...
-      </Switch>
-    </BrowserRouter>
-  </Provider>
-);
+type AppProps = {
+  ssr: ?boolean,
+  url: ?string,
+};
+
+const Base = ({ ssr = false, url }: AppProps) => {
```
Comment (huwd) [inline src/base.jsx]: So we still want the default to be not SSR then?
If generally most apps are gonna need personalisation and sharing, wouldn't that mean a default to ssr is more likely useful?

File: srv/index.js (hunk 4 above)
Comment (Gisleburt) [reply srv/index.js]: No but we can and should make one.
Currently the site uses https://apolitical.co/wp-content/themes/apolitical/public/favicons/favicon-32x32.png

**Comment by huwd** [comment]
oops, got some flow and unit test failures

File: src/base.jsx (hunk 5 above)
Comment (huwd) [reply src/base.jsx]: cool!

**Pull request merged**